src/main/java/com/github/discvrseq/walkers/VcfToLuceneIndexer.java

```diff
@@ -0,0 +1,196 @@
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Annotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.InfoFieldAnnotation;
+
+import com.github.discvrseq.tools.DiscvrSeqProgramGroup;
+import com.github.discvrseq.walkers.annotator.DiscvrVariantAnnotator;
+
+import htsjdk.variant.variantcontext.VariantContext;
+
+/**
+ * This tool accepts a VCF, iterates the variants and writes the results to a lucene search index.
+ * It accepts a list of INFO annotations that will be indexed per site.
+ *
+ * <h3>Usage example:</h3>
+ * <pre>
+ *  java -jar DISCVRseq.jar VcfToLuceneIndexer
+ *     -V myVcf.gz \
+ *     -O /directory/for/output/
+ *     -IF AF
+ *     -AN SampleList
+ * </pre>
+ */
+@DocumentedFeature
+@CommandLineProgramProperties(
+        summary = "This tool accepts a VCF, iterates the variants, and writes the results to a lucene search index",
+        oneLineSummary = "Index a VCF using Apache Lucene",
+        programGroup = DiscvrSeqProgramGroup.class
+)
+public class VcfToLuceneIndexer extends VariantWalker {
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output file (if not provided, defaults to STDOUT)", common = false, optional = true)
+    private File outDir = null;
+
+    @Argument(doc="Info fields to index", fullName = "info-field", shortName = "IF", optional = true)
+    List<String> infoFieldsToIndex;
+
+    @Argument(doc="Annotations to apply", fullName = "annotation-name", shortName = "AN", optional = true)
+    List<String> annotationClassNames;
+
+    IndexWriter writer;
+
+    @Override
+    public List<? extends CommandLinePluginDescriptor<?>> getPluginDescriptors() {
+        // This will eventually be used if we need to implement custom InfoFieldAnnotation classes
+        return Collections.singletonList(new DiscvrVariantAnnotator.DiscvrAnnotationPluginDescriptor());
+    }
+
+    @Override
+    public boolean useVariantAnnotations() {
+        return true;
+    }
+
+    private final List<InfoFieldAnnotation> annotationsToUse = new ArrayList<>();
+
+    @Override
+    public void onTraversalStart() {
+        StandardAnalyzer analyzer = new StandardAnalyzer();
+
+        Directory index;
+        try {
+            index = FSDirectory.open(outDir.toPath());
+        } catch(IOException e) {
+            analyzer.close();
+            throw new GATKException(e.getMessage(), e);
+        }
+
+        IndexWriterConfig config = new IndexWriterConfig(analyzer);
+
+        try {
+            writer = new IndexWriter(index, config);
+        } catch (IOException e) {
+            throw new GATKException(e.getMessage(), e);
+        }
+
+        // This is where you'd sanity check the set of annotations is valid, etc.
+        // Note: we can implement custom InfoFieldAnnotation classes for operations like indexing all
```

Comment:
We should actually iterate the values for "IF", inspect the VcfHeader, and make sure they exist. The header has a data dictionary that tells us the datatype for each field and also whether it is per-site, or one-per-alternative allele